build/dependencies.yaml

```diff
@@ -109,7 +109,7 @@ dependencies:
 
   # From https://github.com/etcd-io/etcd/blob/main/Makefile
   - name: "golang: etcd release version"
-    version: 1.19.9 # https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md
+    version: 1.20.7 # https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md
```

Comment:
https://github.com/etcd-io/etcd/blob/main/CHANGELOG/CHANGELOG-3.5.md#dependencies
Etcd v3.5.10: Compile binaries using https://github.com/etcd-io/etcd/pull/15822. v3.5.10 is the next release.

Currently, etcd in k/k is v3.5.9.

But this should be OK.